src/test/java/com/puppycrawl/tools/checkstyle/PackageObjectFactoryTest.java

```diff
@@ -493,6 +493,32 @@ public void testCreateObjectWithNameContainingPackageSeparator() throws Exceptio
             .isInstanceOf(MockClass.class);
     }
 
+    /**
+    * This test case is designed to verify the behavior of the PackageObjectFactory's
+    * createModule method when it is provided with a class name containing a package separator.
+    * It ensures that the method correctly handles such input and returns an instance of the
+    * specified class, even when external module retrieval results in an exception.
+    * This test case will helpful to save the time complexity
+    */
+    @Test
+    public void testCreateObjectWithNameContainingPackageSeparator2() throws Exception {
+        final ClassLoader classLoader = ClassLoader.getSystemClassLoader();
+        final Set<String> packages = Collections.singleton(BASE_PACKAGE);
+        final PackageObjectFactory objectFactory =
+            new PackageObjectFactory(packages, classLoader, TRY_IN_ALL_REGISTERED_PACKAGES);
+
+        try (MockedStatic<ModuleReflectionUtil> utilities =
+                     mockStatic(ModuleReflectionUtil.class)) {
+            utilities.when(() -> ModuleReflectionUtil.getCheckstyleModules(packages, classLoader))
+                    .thenThrow(new IllegalStateException("mock exception"));
```

Comment:
thenThrow(new IllegalStateException("creation of objects by fully qualified names Shou not call for search of modules in classpath"))

Reply:
done